ctx.go

```diff
@@ -369,16 +369,22 @@ func decoderBuilder(parserConfig ParserConfig) interface{} {
 
 // BodyParser binds the request body to a struct.
 // It supports decoding the following content types based on the Content-Type header:
-// application/json, application/xml, application/x-www-form-urlencoded, multipart/form-data
+// application/json, application/problem+json, application/xml, application/x-www-form-urlencoded, multipart/form-data
 // If none of the content types above are matched, it will return a ErrUnprocessableEntity error
 func (c *Ctx) BodyParser(out interface{}) error {
 	// Get content-type
 	ctype := utils.ToLower(c.app.getString(c.fasthttp.Request.Header.ContentType()))
 
 	ctype = utils.ParseVendorSpecificContentType(ctype)
 
-	// Parse body accordingly
-	if strings.HasPrefix(ctype, MIMEApplicationJSON) {
+    var ctypeEnd int
+    if ctypeEnd = strings.IndexByte(ctype, ';'); ctypeEnd == -1 {
+      ctypeEnd = len(ctype)
+    }
+    isJSON := strings.HasSuffix(utils.TrimRight(ctype, ' '), "+json")
```

Comment:
Wouldn't hasPrefix with "+json" and "/json" be enough?

is ctypeEnd unused ?

can you please share the benchmark numbers before this customization and after, so we can see how much more time or performance gain this customization has

Reply:
I've modified this to discard the rest of the content type from ';' onwards if such section exists. Then we just ensure the suffix of the content type matches "json" since this will be valid in the case of regular "application/json" and JSON extension types such as "application/problem+json".

Here are the benchmark numbers I've recieved after running `go test -v -run=^$ -bench=Benchmark_Ctx_BodyParser_ -benchmem -count=4`:

Before changes:

    goos: linux
    goarch: amd64
    pkg: github.com/gofiber/fiber/v2
    cpu: Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz
    Benchmark_Ctx_BodyParser_JSON
    Benchmark_Ctx_BodyParser_JSON-12                 1604654               879.7 ns/op         240 B/op           6 allocs/op
    Benchmark_Ctx_BodyParser_JSON-12                 1547250               793.3 ns/op         240 B/op           6 allocs/op
    Benchmark_Ctx_BodyParser_JSON-12                 1506621               790.8 ns/op         240 B/op           6 allocs/op
    Benchmark_Ctx_BodyParser_JSON-12                 1512298               731.4 ns/op         240 B/op           6 allocs/op
    Benchmark_Ctx_BodyParser_CustomJSON
    Benchmark_Ctx_BodyParser_CustomJSON-12           1406355               841.5 ns/op         264 B/op           7 allocs/op
    Benchmark_Ctx_BodyParser_CustomJSON-12           1418983               830.3 ns/op         264 B/op           7 allocs/op
    Benchmark_Ctx_BodyParser_CustomJSON-12           1398556              1012 ns/op           264 B/op           7 allocs/op
    Benchmark_Ctx_BodyParser_CustomJSON-12           1515750               843.2 ns/op         264 B/op           7 allocs/op
    Benchmark_Ctx_BodyParser_XML
    Benchmark_Ctx_BodyParser_XML-12                   342775              3827 ns/op          1160 B/op          24 allocs/op
    Benchmark_Ctx_BodyParser_XML-12                   379686              3471 ns/op          1160 B/op          24 allocs/op
    Benchmark_Ctx_BodyParser_XML-12                   511072              2807 ns/op          1160 B/op          24 allocs/op
    Benchmark_Ctx_BodyParser_XML-12                   383931              3299 ns/op          1160 B/op          24 allocs/op
    Benchmark_Ctx_BodyParser_Form
    Benchmark_Ctx_BodyParser_Form-12                 1000000              1220 ns/op           360 B/op          13 allocs/op
    Benchmark_Ctx_BodyParser_Form-12                  868200              1406 ns/op           360 B/op          13 allocs/op
    Benchmark_Ctx_BodyParser_Form-12                  724956              1471 ns/op           360 B/op          13 allocs/op
    Benchmark_Ctx_BodyParser_Form-12                  933985              1190 ns/op           360 B/op          13 allocs/op
    Benchmark_Ctx_BodyParser_MultipartForm
    Benchmark_Ctx_BodyParser_MultipartForm-12        1000000              1071 ns/op           328 B/op          12 allocs/op
    Benchmark_Ctx_BodyParser_MultipartForm-12         887564              1135 ns/op           328 B/op          12 allocs/op
    Benchmark_Ctx_BodyParser_MultipartForm-12        1000000              1066 ns/op           328 B/op          12 allocs/op
    Benchmark_Ctx_BodyParser_MultipartForm-12        1008379              1232 ns/op           328 B/op          12 allocs/op
    PASS
    ok      github.com/gofiber/fiber/v2     34.578s

After changes:

    goos: linux
    goarch: amd64
    pkg: github.com/gofiber/fiber/v2
    cpu: Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz
    Benchmark_Ctx_BodyParser_JSON
    Benchmark_Ctx_BodyParser_JSON-12                 1840669               795.7 ns/op         240 B/op           6 allocs/op
    Benchmark_Ctx_BodyParser_JSON-12                 1473236               782.8 ns/op         240 B/op           6 allocs/op
    Benchmark_Ctx_BodyParser_JSON-12                 1431157               767.5 ns/op         240 B/op           6 allocs/op
    Benchmark_Ctx_BodyParser_JSON-12                 1541245               852.0 ns/op         240 B/op           6 allocs/op
    Benchmark_Ctx_BodyParser_CustomJSON
    Benchmark_Ctx_BodyParser_CustomJSON-12           1471304               741.8 ns/op         264 B/op           7 allocs/op
    Benchmark_Ctx_BodyParser_CustomJSON-12           1410648               910.7 ns/op         264 B/op           7 allocs/op
    Benchmark_Ctx_BodyParser_CustomJSON-12           1298179               809.0 ns/op         264 B/op           7 allocs/op
    Benchmark_Ctx_BodyParser_CustomJSON-12           1612112               809.6 ns/op         264 B/op           7 allocs/op
    Benchmark_Ctx_BodyParser_XML
    Benchmark_Ctx_BodyParser_XML-12                   420786              2801 ns/op          1160 B/op          24 allocs/op
    Benchmark_Ctx_BodyParser_XML-12                   496807              3088 ns/op          1160 B/op          24 allocs/op
    Benchmark_Ctx_BodyParser_XML-12                   411742              2764 ns/op          1160 B/op          24 allocs/op
    Benchmark_Ctx_BodyParser_XML-12                   476277              2951 ns/op          1160 B/op          24 allocs/op
    Benchmark_Ctx_BodyParser_Form
    Benchmark_Ctx_BodyParser_Form-12                  927964              1315 ns/op           360 B/op          13 allocs/op
    Benchmark_Ctx_BodyParser_Form-12                  937864              1303 ns/op           360 B/op          13 allocs/op
    Benchmark_Ctx_BodyParser_Form-12                  729925              1434 ns/op           360 B/op          13 allocs/op
    Benchmark_Ctx_BodyParser_Form-12                 1008754              1242 ns/op           360 B/op          13 allocs/op
    Benchmark_Ctx_BodyParser_MultipartForm
    Benchmark_Ctx_BodyParser_MultipartForm-12         919500              1156 ns/op           328 B/op          12 allocs/op
    Benchmark_Ctx_BodyParser_MultipartForm-12        1000000              1201 ns/op           328 B/op          12 allocs/op
    Benchmark_Ctx_BodyParser_MultipartForm-12         836179              1303 ns/op           328 B/op          12 allocs/op
    Benchmark_Ctx_BodyParser_MultipartForm-12        1237518              1047 ns/op           328 B/op          12 allocs/op
    PASS
    ok      github.com/gofiber/fiber/v2     34.456s

Reply:
Need to discard params like charset `application/json; charset=utf-8`

Reply:
I believe the updated code should do this. We now index the byte of ';' (if found) and take a string slice up to and excluding this byte.